lib/checks/color/color-contrast-evaluate.js

```diff
@@ -188,4 +148,52 @@ function colorContrastEvaluate(node, options, virtualNode) {
   return isValid;
 }
 
-export default colorContrastEvaluate;
+function findPseudoElement(vNode, {
+  pseudoSizeThreshold = 0.25,
+  ignorePseudo = false
+}) {
+  if (ignorePseudo) {
+    return;
+  }
+  const rect = vNode.boundingClientRect;
+  const minimumSize = rect.width * rect.height * pseudoSizeThreshold;
+  do {
+    const beforeSize = getPseudoElementArea(vNode.actualNode, ':before')
+    const afterSize = getPseudoElementArea(vNode.actualNode, ':after')
+    if (beforeSize + afterSize > minimumSize) {
+      return vNode
+    }
+  } while (vNode = vNode.parent);
+}
+
+const getPseudoElementArea = memoize(function hasPseudoElement(node, pseudo) {
+  const style = window.getComputedStyle(node, pseudo);
+  const matchPseudoStyle = (prop, value) => style.getPropertyValue(prop) === value;
+  if (
+    matchPseudoStyle('content', 'none') ||
+    matchPseudoStyle('display', 'none') ||
+    matchPseudoStyle('visibility', 'hidden') ||
+    matchPseudoStyle('position', 'absolute') === false
+  ) {
+    return 0; // The pseudo element isn't visible
+  }
+
+  if (
+    getOwnBackgroundColor(style).alpha === 0 &&
+    matchPseudoStyle('background-image', 'none')
+  ) {
+    return 0; // There is no background
+  }
+
+  // Find the size of the pseudo element;
+  const pseudoWidth = parseInt(style.getPropertyValue('width'));
+  const pseudoHeight = parseInt(style.getPropertyValue('height'));
```

Comment:
TODO: Check the unit here. If this isn't 'px', return Invinity. IE doesn't normalise units. They'll just have to review them.